src/classes/Addresses.cls

```diff
@@ -0,0 +1,1133 @@
+                        dmlWrapperLocal, false);
+        if (errors.errorsExist) {
+            putErrorsOnAddresses(errors, defaultAndSeasonalWithChanges(), nonDefaultsWithChanges());
+        }
+    }
+
+    public void onAfterDelete() {
+        if (!addressesByHouseholdAccountId().isEmpty()) {
+            refreshCurrentHHAddress(addressesByHouseholdAccountId(), dmlWrapperLocal,
+                    false);
+        }
+
+        // we must process dmlWrapperLocal ourselves, so we can extract out the appropriate errors
+        // that might occur updating contacts and accounts, and associating those errors with
+        // the correct addresses.
+        ERR_Handler.Errors errors = TDTM_TriggerHandler.processDMLAllOrNothing(dmlWrapperLocal,
+                false);
+        if (errors.errorsExist) {
+            putErrorsOnAddresses(errors, addressesByHouseholdAccountId(), new Map<Id, Address__c>());
+        }
+    }
+
+    /********************************************************************************************************
+    * @description Retreive the defined default address type picklist value to use when creating a new
+    * Address record.
+    */
+    public static String defaultAddressType {
+        get {
+            if (defaultAddressType == null) {
+                for (Schema.PicklistEntry pv: Address__c.Address_Type__c.getDescribe().getPicklistValues()) {
+                    if (pv.isDefaultValue()) {
+                        defaultAddressType = pv.getValue();
+                        break;
+                    }
+                }
+            }
+            return defaultAddressType;
+        } private set;
+    }
+
+    public Map<Id, Address__c> addressesByHouseholdAccountId() {
+        Map<Id, Address__c> mapAddressesByAcctId = new Map<Id, Address__c>(); // map of HH Accounts, and their default Address
+        for (Address__c address : (List<Address__c>) getRecords()) {
+            if (hasHouseholdAccount(address)) {
+                mapAddressesByAcctId.put(householdAccountIdFor(address), null);
+            }
+        }
+        return mapAddressesByAcctId;
+    }
+
+    /*******************************************************************************************************
+    * @description utility to compare a Contact or Account address to the Address record
+    * @param sObj Account or Contact
+    * @param addr Address
+    * @return boolean. true if any of the Address fields on the Contact are different from this Address record
+    */
+    public static Boolean shouldUpdateAddress(SObject sObj, Address__c addr) {
+        if (sObj == null || addr == null) {
+            return false;
+        }
+        Boolean isContact = (sObj.getSObjectType() == Contact.SObjectType);
+        String prefix = (isContact ? 'Mailing' : 'Billing');
+
+        Boolean isDifferent = (
+                !equalsCaseSensitive((String) sObj.get(prefix + 'Street'), getMultilineStreet(addr)) ||
```

Comment:
This looks like another spot that we'd want to push down into the NPSP Address Entity.

Reply:
Yep, great call.  I just moved things around a bit.  Check out the latest commit and let me know what you think.